chapter_linear-regression/oo-design.md

```diff
@@ -162,26 +174,49 @@ Sometimes we put the code to compute the output into a separate `forward` method
 ```{.python .input}
 %%tab all
 class Module(d2l.nn_Module, d2l.HyperParameters):  #@save
-    def __init__(self, plot_train_per_epoch=2, plot_valid_per_epoch=1):
-        super().__init__()
-        self.save_hyperparameters()
-        self.board = ProgressBoard()
+    if tab.selected('pytorch', 'mxnet', 'tensorflow'):
+        def __init__(self, plot_train_per_epoch=2, plot_valid_per_epoch=1):
+            super().__init__()
+            self.save_hyperparameters()
+            self.board = ProgressBoard()
         if tab.selected('tensorflow'):
             self.training = None
 
+    if tab.selected('jax'):
+        # No need for save_hyperparam when using python dataclass
```

Comment:
Python

Reply:
Add a period to the end of a sentence if this sentence is not the end of the comment. Don't add multiple comments for the same line of code: just merge them into a paragraph of multiple sentences.

Reply:
Cool, then we can just leave this as is because I removed the python dataclass comment in the last commit since it was confusing for the readers.